pyfar/dsp/filter/fractional_octaves.py

```diff
@@ -971,8 +971,8 @@ def _encode(self):
     def _decode(cls, obj_dict):
         # initialize new class instance
         obj = cls(obj_dict["_num_fractions"], obj_dict["_frequency_range"],
-                  obj_dict["_sampling_rate"], obj_dict["_overlap"],
-                  obj_dict["_slope"], obj_dict["_n_samples"])
+                  obj_dict["_overlap"], obj_dict["_slope"],
+                  obj_dict["_n_samples"], obj_dict["_sampling_rate"])
```

Comment:
`"_state"` and `"_initialized"` must also be added. Otherwise they would be lost when they do not have their default values. This must probably also be done for the reconstructing filter bank

Reply:
Does that mean that the `FractionalOctaveBands` needs [this](https://github.com/jake-is-ESD-protected/pyfar/blob/bf187e53bdf55a9aa118ddc68f349345933bf412/pyfar/dsp/filter/gammatone.py#L388-L411) implementation? As for `_initialized`, where is that used? I can't find it in `GammatoneBands` as a reference.

Reply:
Its already implemented in the filter-class from which you inherit. So no need to implement it here, but you need to save the parameters `"_state"` and `"_initialized"` , and use them for loading class objects. Does that make it more clear?